packages/workflow/test/Workflow.test.ts

```diff
@@ -2044,4 +2020,368 @@ describe('Workflow', () => {
+				},
+			};
+
+			const workflow = new Workflow({
+				id: 'test',
+				nodes: [node1, node2, targetNode],
+				connections,
+				active: false,
+				nodeTypes,
+			});
+
+			const result = workflow.getHighestNode(targetNode.name);
+			expect(result).toEqual([node2.name]);
+		});
+
+		test('should handle nested connections', () => {
+			const node1 = createNode('Node1');
+			const node2 = createNode('Node2');
+			const node3 = createNode('Node3');
+			const targetNode = createNode('TargetNode');
+
+			const connections = {
+				Node3: {
+					main: [
+						[{ node: 'Node1', type: NodeConnectionType.Main, index: 0 }],
+						[{ node: 'Node2', type: NodeConnectionType.Main, index: 0 }],
+					],
+				},
+				TargetNode: {
+					main: [[{ node: 'Node3', type: NodeConnectionType.Main, index: 0 }]],
+				},
+			};
+
+			const workflow = new Workflow({
+				id: 'test',
+				nodes: [node1, node2, node3, targetNode],
+				connections,
+				active: false,
+				nodeTypes,
+			});
+
+			const result = workflow.getHighestNode(targetNode.name);
+			expect(result).toEqual([targetNode.name]);
+		});
+
+		test('should respect specified connection type', () => {
+			const node1 = createNode('Node1');
+			const node2 = createNode('Node2');
+			const targetNode = createNode('TargetNode');
+
+			const connections = {
+				Node1: {
+					main: [[{ node: 'TargetNode', type: NodeConnectionType.Main, index: 0 }]],
+				},
+			};
+
+			const workflow = new Workflow({
+				id: 'test',
+				nodes: [node1, node2, targetNode],
+				connections,
+				active: false,
+				nodeTypes,
+			});
+
+			const resultMain = workflow.getHighestNode(targetNode.name, NodeConnectionType.Main);
+
+			expect(resultMain).toEqual([node1.name]);
+		});
+
+		test('should handle specified connection index', () => {
+			const node1 = createNode('Node1');
+			const node2 = createNode('Node2');
+			const targetNode = createNode('TargetNode');
+
+			const connections = {
+				Node1: {
+					main: [[{ node: 'TargetNode', type: NodeConnectionType.Main, index: 0 }]],
+				},
+				Node2: {
+					main: [[], [{ node: 'TargetNode', type: NodeConnectionType.Main, index: 1 }]],
+				},
+			};
+
+			const workflow = new Workflow({
+				id: 'test',
+				nodes: [node1, node2, targetNode],
+				connections,
+				active: false,
+				nodeTypes,
+			});
+
+			const resultFirstIndex = workflow.getHighestNode(targetNode.name, NodeConnectionType.Main, 0);
+			const resultSecondIndex = workflow.getHighestNode(
+				targetNode.name,
+				NodeConnectionType.Main,
+				1,
+			);
+
+			expect(resultFirstIndex).toEqual([node1.name]);
+			expect(resultSecondIndex).toEqual([node2.name]);
+		});
+
+		test('should prevent infinite loops with cyclic connections', () => {
+			const node1 = createNode('Node1');
+			const node2 = createNode('Node2');
+			const targetNode = createNode('TargetNode');
+
+			const connections = {
+				Node1: {
+					main: [[{ node: 'Node2', type: NodeConnectionType.Main, index: 0 }]],
```

Comment:
Should we also have a test where a node has multiple connections? E.g. something like

Node1 --> TargetNode
           --> Node2